tests/test_xet_utils.py

```diff
@@ -4,28 +4,40 @@
 
 from huggingface_hub import constants
 from huggingface_hub.utils._xet import (
-    XetMetadata,
-    _fetch_xet_metadata_with_url,
-    parse_xet_headers,
-    refresh_xet_metadata,
+    XetFileData,
+    _fetch_xet_connection_info_with_url,
+    parse_xet_connection_info_from_headers,
+    parse_xet_file_data_from_headers,
+    refresh_xet_connection_info,
 )
 
 
-def test_parse_valid_headers_minimal() -> None:
+def test_parse_valid_headers_file_info() -> None:
+    headers = {
+        "X-Xet-Hash": "sha256:abcdef",
+        "X-Xet-Refresh-Route": "/api/refresh",
+    }
+
+    file_data = parse_xet_file_data_from_headers(headers)
+
+    assert file_data is not None
+    assert file_data.refresh_route == "/api/refresh"
+    assert file_data.file_hash == "sha256:abcdef"
```

Comment:
```suggestion
    assert file_data.file_hash == "sha256:abcdef"


def test_parse_invalid_headers_file_info() -> None:
    assert parse_xet_file_data_from_headers({"X-foo": "bar"}) is None
```

Would be good to test on empty headers that it doesn't fail (and return None instead)